go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -895,6 +895,14 @@ func (npw *nodePortWatcher) UpdateService(old, new *corev1.Service) error {
 		if err = delServiceRules(old, svcConfig.localEndpoints, npw); err != nil {
 			errors = append(errors, err)
 		}
+
+		// Delete conntrack entries for UDP ports of the service if they have changed.
+		// UDP is connectionless, so stale conntrack entries can cause packets to be routed to
+		// old target ports even after service updates. Unlike TCP (which is stateful and manages
+		// its own connection lifecycle), UDP conntrack entries must be explicitly cleaned up.
+		if err = npw.deleteConntrackForUpdatedUDPServicePorts(old, new); err != nil {
```

Comment:
Can we check if the service ip/port changed and just invoke `deleteConntrackForService(old)` instead? Even if some conntrack entries are removed unintentionally, the NEW packet will match the same rule as the original packet did. The impact on traffic should be minimal in the worst case.